src/lsp/cobol_parser/grammar_tokens.mly

```diff
@@ -1399,4 +1399,23 @@
 %token XML_NTEXT                    [@keyword.silenced]
 %token XML_TEXT                     [@keyword.silenced]
 
+(* INTRINSIC FUNCTIONS *)
+%token <string> INTRINSIC_FUNC
```

Comment:
Here I'd go with a `%token <Cobol_ptree.intrinsic_function_identifier> INTRINSICS_FUNC`.

Reply:
But for the case of a `CallFunc` we need a `name with_loc`, I can change it to `UserDefined of name with_loc | Intrinsic of intrinsic_function_identifier`

Reply:
What is this token used for actually? I would have thought it's something like a `used-defined-intrinsic-name` or something like that. If it's user-defined, the scanner/tokenizer should have the mapping to the actual intrinsic token already, and therefore this one wouldn't be useful.

Reply:
It's used in `function_identifier` rule of the grammar to allow not having `FUNCTION` in front of the intrinsic name if the name was specified in the REPOSITORY. Wither we have this token, for intrinsic function without special handling of arguments.
If we don't have this token, we should have one token by intrinsic function name, which just adds more matching in the grammar to handle this special case

Reply:
I see. So this should be something like `GENERIC_INTRINSIC`, not carrying a name, but with a piece of AST that references the intrinsic. These identifiers should all be known in advance, so we can have a dedicated ADT for them.